nav2_util/src/motion_model/motion_model.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_util/motion_model/motion_model.hpp"
+
+#include <string>
+
+namespace nav2_util
+{
+
+MotionModel *
+MotionModel::createMotionModel(
+  std::string & type, double alpha1, double alpha2,
```

Comment:
Shouldn't this be an enumerated type so the compiler can validate it at compile time instead of a string?

Reply:
It could be. This was simply code factored out of AMCL. The type string originates as a parameter to AMCL. So, it was probably convenient to simply do a string compare. We could change it, but I'd prefer outside of this PR.

Reply:
OK, what if type is something invalid, right now this function won't return any value. Shouldn't it at least return the DifferentialMotionModel by default? If we're not going to fix this now, can we file a bug to fix it?